src/sentry/snuba/metrics/query_builder.py

```diff
@@ -119,6 +119,12 @@ def resolve_tags(org_id: int, input_: Any) -> Any:
         if input_.function == "ifNull":
             # This was wrapped automatically by QueryBuilder, remove wrapper
             return resolve_tags(org_id, input_.parameters[0])
+        elif input_.function == "isNull":
+            return Condition(
+                lhs=resolve_tags(org_id, input_.parameters[0]),
+                op=Op.EQ,
+                rhs=resolve_tags(org_id, ""),
+            )
```

Comment:
Does this also work for nested conditions like `release:foo or release:""`. I think that might resolve to `Function("or", [Function("equals", ...), ..]` and then you get a `Condition` as a function argument.

Reply:
Thanks, fixed. I believe we should essentially never convert Function to Condition or vice-versa in resolve_tags (though we already do)

Reply:
I agree, what other conversion instance is there? I can't seem to spot it.

Reply:
potentially line 121, though i believe conversion between "expressions" and functions might be fine